File: 2017/1-inverse-captcha/1.py
#!/usr/bin/env python3


def get_challenge():
    with open("./input", "r") as pifh:
        line = pifh.readline()
    return line.strip()


def solve(puzzle_input):
    value = 0
    next_digit = None
    for index, digit in enumerate(puzzle_input):
        digit = int(digit)
        try:
            next_digit = int(puzzle_input[index + 1])
        except IndexError:
            next_digit = int(puzzle_input[0])
        if digit == next_digit:
            value += digit
    return value


print(solve("1122") == 3)
print(solve("1111") == 4)
print(solve(get_challenge()))

File: 2017/01-inverse-captcha/1.go
package main

import (
	"fmt"
    "io/ioutil"
    "strings"
)

func GetChallenge() string {
    filename := "./input"
    b, _ := ioutil.ReadFile(filename)
    return strings.TrimSpace(string(b))
}

func Solve(s string) (x int) {
    s_len := len(s)
    for c := 0; c < s_len; c++ {
        digit := int(s[c])
        next_digit := -1
        if (c + 1) < s_len {
            next_digit = int(s[c + 1])
        } else {
            next_digit = int(s[0])
        }
        if digit == next_digit {
            x += (digit - 48)
        }
    }
    return
}

func main() {
    fmt.Println(Solve("1122") == 3)
    fmt.Println(Solve("1111") == 4)
    fmt.Println(Solve(GetChallenge()))
}
